src/transxchange/read.rs

```diff
@@ -59,6 +124,19 @@ fn load_company(operator: &Element) -> Result<Company> {
 }
 
 fn read_transxchange(transxchange: &Element, collections: &mut Collections) -> Result<()> {
+    let service_validity_period = get_service_validity_period(transxchange)?;
+    let dataset_indexes: Vec<_> = collections
+        .datasets
+        .get_id_to_idx()
+        .values()
+        .cloned()
+        .collect();
+    dataset_indexes.into_iter().try_for_each(|index| {
+        update_validity_period(
+            &mut collections.datasets.index_mut(index),
+            &service_validity_period,
+        )
+    })?;
```

Comment:
you could do "set min and max` + get_service_validity_period + update_validity_period in a same function

Reply:
I'm not sure we can because the "set min/max" is done only once, and the rest is done on every TransXChange file. But I can take a look about merging `get_service_validity_period` and `update_validity_period`.

Reply:
I extracted a function `update_validity_period_from_transxchange` for the logic of updating the datasets. Inside, I keep calling the existing functions `get_service_validity_period` and `update_validity_period` because each is really doing something different (easier to test too if they're independent). The new function is wrapping the calls to these functions (see 6245d6fc146fd845f6f572bd97778ac2752443b6).